frontend/src/metabase/dashboard/containers/CreateDashboardModal.unit.spec.tsx

```diff
@@ -17,32 +17,14 @@ const ROOT_COLLECTION = {
   can_write: true,
 } as Collection;
 
-function mockCachingEnabled(enabled = true) {
-  const original = MetabaseSettings.get.bind(MetabaseSettings);
-  const spy = jest.spyOn(MetabaseSettings, "get");
-  spy.mockImplementation(key => {
-    if (key === "enable-query-caching") {
-      return enabled;
-    }
-    if (key === "application-name") {
-      return "Metabase Test";
-    }
-    if (key === "version") {
-      return { tag: "" };
-    }
-    if (key === "is-hosted?") {
-      return false;
-    }
-    if (key === "enable-enhancements?") {
-      return false;
-    }
-    return original(key);
-  });
-}
-
-function setup({ mockCreateDashboardResponse = true } = {}) {
+function setup({
+  isCachingEnabled = false,
+  mockCreateDashboardResponse = true,
+} = {}) {
   const onClose = jest.fn();
 
+  const settings = mockSettings({ "enable-query-caching": isCachingEnabled });
```

Comment:
Similar to the comment above, where it's not apparent to me how we decide what keys we want to keep for mocking from the previous version of the code.

Reply:
Answered above